src/ray/raylet/node_manager.cc

```diff
@@ -522,7 +522,11 @@ void NodeManager::DispatchTasks(const std::list<Task> &ready_tasks) {
     if (!local_available_resources_.Contains(task_resources)) {
       // Not enough local resources for this task right now, skip this task.
       // TODO(rkn): We should always skip node managers that have 0 CPUs.
-      continue;
+      if (can_skip) {
```

Comment:
This if-block is a bit confusing...if we're going to keep this, I think we at least need a better name (maybe something like `identical_resources`) and/or a comment here explaining why this logic is here.

Reply:
I documented it in the header:
```
  /// \param can_skip If true, all the tasks in ready_tasks will have the same
  ///        resource requirements, so we can stop processing them once one of
  ///        them is infeasible.
```
But can try to make it clearer if it currently is not, let me know :)